libpod/container_internal.go

```diff
@@ -271,6 +272,43 @@ func (c *Container) export(path string) error {
 	return err
 }
 
+func (c *Container) addImageVolumes() error {
+	mountPoint := c.state.Mountpoint
+	if !c.state.Mounted {
+		mount, err := c.runtime.store.Mount(c.ID(), c.config.MountLabel)
+		if err != nil {
+			return errors.Wrapf(err, "error mounting container %q", c.ID())
+		}
+		mountPoint = mount
+		defer func() {
+			if err := c.runtime.store.Unmount(c.ID()); err != nil {
+				logrus.Errorf("error unmounting container %q: %v", c.ID(), err)
+			}
+		}()
+	}
+
+	for i, vol := range c.config.Spec.Mounts {
```

Comment:
Instead of looping over mounts to find those with no source, we should look up the image's volumes here and add them to spec.Mounts, maybe with a check to make sure they don't override user mounts

Reply:
To be more specific, if we put this in Init() when we're generating the spec, we can grab the image data here and get the volumes from it. We loop over the volumes, first checking if the spec already has a volume with that destination (if it does ignore), and then adding the volume to the spec and creating the source like we do here.

Reply:
I agree.

Reply:
fixed